pkgs/io_file/lib/src/vm_windows_file_system.dart

```diff
@@ -80,4 +165,118 @@ base class WindowsFileSystem extends FileSystem {
       throw _getError(errorCode, 'rename failed', oldPath);
     }
   });
+
+  /// Sets metadata for the file system entity.
+  ///
+  /// TODO(brianquinlan): Document the arguments.
+  void setMetadata(
+    String path, {
+    bool? isReadOnly,
+    bool? isHidden,
+    bool? isSystem,
+    bool? isArchive,
+    bool? isTemporary,
+    bool? isContentNotIndexed,
+    bool? isOffline,
+  }) => using((arena) {
+    if ((isReadOnly ??
+            isHidden ??
+            isSystem ??
+            isArchive ??
+            isTemporary ??
+            isContentNotIndexed ??
+            isOffline) ==
+        null) {
+      return;
+    }
+    final fileInfo = arena<win32.WIN32_FILE_ATTRIBUTE_DATA>();
+    final nativePath = path.toNativeUtf16(allocator: arena);
+    if (win32.GetFileAttributesEx(
+          nativePath,
+          win32.GetFileExInfoStandard,
+          fileInfo,
+        ) ==
+        win32.FALSE) {
+      final errorCode = win32.GetLastError();
+      throw _getError(errorCode, 'set metadata failed', path);
+    }
+
+    var attributes = fileInfo.ref.dwFileAttributes;
+    if (attributes == win32.FILE_ATTRIBUTE_NORMAL) {
+      // `FILE_ATTRIBUTE_NORMAL` indicates that no other attributes are set and
+      // is valid only when used alone.
+      attributes = 0;
+    }
+
+    int setBit(int base, int value, bool? bit) => switch (bit) {
+      null => base,
+      true => base | value,
+      false => base & ~value,
+    };
+
+    attributes = setBit(attributes, win32.FILE_ATTRIBUTE_READONLY, isReadOnly);
+    attributes = setBit(attributes, win32.FILE_ATTRIBUTE_HIDDEN, isHidden);
+    attributes = setBit(attributes, win32.FILE_ATTRIBUTE_SYSTEM, isSystem);
+    attributes = setBit(attributes, win32.FILE_ATTRIBUTE_ARCHIVE, isArchive);
+    attributes = setBit(
+      attributes,
+      win32.FILE_ATTRIBUTE_TEMPORARY,
+      isTemporary,
+    );
+    attributes = setBit(
+      attributes,
+      win32.FILE_ATTRIBUTE_NOT_CONTENT_INDEXED,
+      isContentNotIndexed,
+    );
+    attributes = setBit(attributes, win32.FILE_ATTRIBUTE_OFFLINE, isOffline);
+    if (win32.SetFileAttributes(nativePath, attributes) == win32.FALSE) {
```

Comment:
(Should this use `win32.FILE_ATTRIBUTE_NORMAL` if attributes is zero? I think it probably should, but I couldn't find docs that were specific to `SetFileAttributes`. It makes a difference for `CreateFile` where `0` makes it reuse the attributes of an existing file, if there is one, and `...NORMAL` clears the attributes.)

Reply:
The `attributes` can't be zero because the check on Ln 182 ensures that at least one attribute is set.

Reply:
Done.